src/deepagents/sub_agent.py

```diff
@@ -5,17 +5,23 @@
 from typing import TypedDict
 from langchain_core.tools import tool, InjectedToolCallId
 from langchain_core.messages import ToolMessage
-from typing import Annotated, NotRequired
+from typing import Annotated, NotRequired, Optional
 from langgraph.types import Command
 
 from langgraph.prebuilt import InjectedState
+from deepagents.model import get_default_model
 
 
 class SubAgent(TypedDict):
     name: str
     description: str
     prompt: str
     tools: NotRequired[list[str]]
+    # Optional per-subagent model configuration
+    model: NotRequired[str]
+    model_provider: NotRequired[str]
+    max_tokens: NotRequired[int]
```

Comment:
what about adding a model settings dictionary? makes it easier to add more parameters?